### PR TITLE
PR Feedback implementation

### PR DESCRIPTION
**Pull request by Ivor**
This PR implements feedback that was received on the previous [PR23](https://github.com/viodotcom/meta_logger/pull/23).

**Highlights** 🧵 
- https://github.com/viodotcom/meta_logger/pull/24#discussion_r1816791273
- https://github.com/viodotcom/meta_logger/pull/24#discussion_r1816792015
- https://github.com/viodotcom/meta_logger/pull/24#discussion_r1816792385
- https://github.com/viodotcom/meta_logger/pull/24#discussion_r1816792877
- https://github.com/viodotcom/meta_logger/pull/24#discussion_r1816794303
- https://github.com/viodotcom/meta_logger/pull/24#discussion_r1816795364

🎗️ Related PRs
- https://github.com/viodotcom/meta_logger/pull/23

### CODE REVIEW

File: lib/meta_logger/slicer.ex [hunk 1]
```diff
@@ -1,6 +1,6 @@
 defmodule MetaLogger.Slicer do
   @moduledoc """
-  A bebaviour for slicing long entries into a list of entries shorter than a passed `max_entry_length` value.
+  A behaviour for slicing long entries into a list of entries shorter than a passed `max_entry_length` value.
```
Comment (Ivor) [inline lib/meta_logger/slicer.ex:3]: 🧵 https://github.com/viodotcom/meta_logger/pull/23#discussion_r1816695835

File: lib/meta_logger/slicer/default_impl.ex [hunk 2]
```diff
@@ -25,13 +25,12 @@ defmodule MetaLogger.Slicer.DefaultImpl do
 
   """
   @impl MetaLogger.Slicer
-  @spec slice(binary(), MetaLogger.Slicer.max_entry_length()) :: [binary()]
```
Comment (Ivor) [inline lib/meta_logger/slicer/default_impl.ex:28]: :thread: https://github.com/viodotcom/meta_logger/pull/23#discussion_r1816697782

File: lib/meta_logger/slicer/default_impl.ex [hunk 3]
```diff
@@ -25,13 +25,12 @@ defmodule MetaLogger.Slicer.DefaultImpl do
...
   def slice(entry, max_entry_length)
       when max_entry_length == :infinity
       when byte_size(entry) <= max_entry_length,
       do: [entry]
 
-  def slice(entry, max_entry_length) do
+  def slice(entry, max_entry_length) when is_binary(entry) do
```
Comment (Ivor) [inline lib/meta_logger/slicer/default_impl.ex:33]: 🧵 https://github.com/viodotcom/meta_logger/pull/23#discussion_r1816702948

File: lib/meta_logger/slicer/utf8_impl.ex [hunk 4]
```diff
@@ -34,9 +34,8 @@ defmodule MetaLogger.Slicer.Utf8Impl do
       when byte_size(entry) <= max_entry_length,
       do: [entry]
 
-  def slice(entry, max_entry_length) do
-    do_slice(entry, max_entry_length, [], [], 0)
-  end
+  def slice(entry, max_entry_length) when is_binary(entry),
```
Comment (Ivor) [inline lib/meta_logger/slicer/utf8_impl.ex:37]: https://github.com/viodotcom/meta_logger/pull/23#discussion_r1816705022

File: lib/meta_logger/slicer/utf8_impl.ex [hunk 5]
```diff
@@ -78,12 +77,10 @@ defmodule MetaLogger.Slicer.Utf8Impl do
...
-    [reconstruct_current_slice_as_binary(partial_slice) | slices]
-  end
+  defp bank_partial_slice(slices, partial_slice),
+    do: [reconstruct_current_slice_as_binary(partial_slice) | slices]
 
   @spec reconstruct_current_slice_as_binary([iodata()]) :: binary()
-  defp reconstruct_current_slice_as_binary(current_slice) do
-    IO.iodata_to_binary(Enum.reverse(current_slice))
-  end
+  defp reconstruct_current_slice_as_binary(current_slice),
```
Comment (Ivor) [inline lib/meta_logger/slicer/utf8_impl.ex:84]: :thread: https://github.com/viodotcom/meta_logger/pull/23#discussion_r1816705662

File: test/meta_logger/slicer/default_impl_test.exs [hunk 6]
```diff
@@ -18,8 +18,10 @@ defmodule MetaLogger.Slicer.DefaultImplTest do
       assert Subject.slice(entry, :infinity) == [entry]
     end
 
-    test "when max entry length is smaller than the size of given entry, " <>
-           "returns a list with one entry",
+    test """
```
Comment (Ivor) [inline test/meta_logger/slicer/default_impl_test.exs:21]: :thread: https://github.com/viodotcom/meta_logger/pull/23#discussion_r1816707004

File: lib/meta_logger/slicer/utf8_impl.ex [hunk 7]
```diff
@@ -78,12 +77,10 @@ defmodule MetaLogger.Slicer.Utf8Impl do
...
+    do: IO.iodata_to_binary(Enum.reverse(current_slice))
```
Comment (prodis) [inline lib/meta_logger/slicer/utf8_impl.ex:85]:
```suggestion
  defp reconstruct_current_slice_as_binary(current_slice) do
    current_slice
    |> Enum.reverse()
    |> IO.iodata_to_binary()
  end
```

File: test/meta_logger/slicer/utf8_impl_test.exs [hunk 8]
```diff
@@ -33,8 +35,10 @@ defmodule MetaLogger.Slicer.Utf8ImplTest do
       assert Subject.slice(entry, 5) == ["01234", "56789"]
     end
 
-    test "when given max entry length is three and the given entry size is 10, " <>
-           "returns a list with four entries",
+    test """
+         when given max entry length is three and the given entry size is 10,
```
Comment (prodis) [inline test/meta_logger/slicer/utf8_impl_test.exs:39]:
```suggestion
         when given max entry length is three and the given entry size is 10, \
```

**Review by prodis — APPROVED** [review]

**Comment by nvieirafelipe** [comment]
ℹ️ I'm not releasing a new version with those changes as they are only related to the style guide.

**Pull request merged**